test/library/standard/Map/testGetValue.chpl

```diff
@@ -0,0 +1,15 @@
+use Map;
+
+class C {
+  var i: int;
+}
+
+var m: map(string, shared C);
+
+m.add("one", new shared C(1));
+m.add("two", new shared C(2));
+
+var x = m.getValue("one");
```

Comment:
This seems to show that something isn't working right. `getValue` returns `const`, but this stores it into a `var` and then modifies it.

Reply:
Do we even document the `const` return intent today? I am going to go out on a limb here and assume it means "const value" and not shorthand for "const ref"...

If that's the case I would expect the compiler to make another copy when moving the tmp at the callsite into the variable.

I'll go ahead and add a test to verify that.